sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/compatibility/SpringCloudAzureSpringBootVersionVerifier.java

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.compatibility;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.SpringBootVersion;
+import org.springframework.util.StringUtils;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+class SpringCloudAzureSpringBootVersionVerifier {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SpringCloudAzureSpringBootVersionVerifier.class);
+    final Map<String, CompatibilityPredicate> supportedVersions = new HashMap<String, CompatibilityPredicate>() {
+        {
+            this.put("2.5", SpringCloudAzureSpringBootVersionVerifier.this.is2_5());
+            this.put("2.6", SpringCloudAzureSpringBootVersionVerifier.this.is2_6());
+        }
+    };
+    private final List<String> acceptedVersions;
+
+    SpringCloudAzureSpringBootVersionVerifier(List<String> acceptedVersions) {
+        this.acceptedVersions = acceptedVersions;
+    }
+
+    /**
+     * Verify the current spring-boot version
+     * @return Verification result of spring-boot version
+     * @throws SpringCloudAzureCompatibilityNotMetException thrown if using an unsupported spring-boot version
+     */
+    public VerificationResult verify() {
+        if (this.springBootVersionMatches()) {
+            return VerificationResult.compatible();
+        } else {
+            List<VerificationResult> errors = new ArrayList<VerificationResult>(Collections.singleton(VerificationResult.notCompatible(this.errorDescription(), this.action())));
+            throw new SpringCloudAzureCompatibilityNotMetException(errors);
+        }
+    }
+
+    private String errorDescription() {
+        String versionFromManifest = this.getVersionFromManifest();
+        return StringUtils.hasText(versionFromManifest) ? String.format("Spring Boot [%s] is not compatible with this Spring Cloud Azure release train", versionFromManifest) : "Spring Boot is not compatible with this Spring Cloud Azure release train";
+    }
+
+    private String action() {
+        return String.format("Change Spring Boot version to one of the following versions %s .%nYou can find the latest Spring Boot versions here [%s]. %nIf you want to learn more about the Spring Cloud Azure Release train compatibility, you can visit this page [%s] and check the [Release Trains] section.%nIf you want to disable this check, "
+            + "just set the property [spring.cloud.azure.compatibility-verifier.enabled=false]", this.acceptedVersions, "https://spring.io/projects/spring-boot#learn", "https://github.com/Azure/azure-sdk-for-java/wiki/Spring-Versions-Mapping");
```

Comment:
how about separate the String to several lines to reduce the line length and improve the readability of the source code.